index.html

```diff
@@ -195,69 +335,537 @@ <h2>
+                    <ol>
+                      <li>Set
+                      |gamepad|.{{Gamepad/[[buttonMapping]]}}[|rawInputIndex|]
+                      to |canonicalIndex|.
+                      </li>
+                      <li>[=list/Append=] |canonicalIndex| to
+                      |mappedIndexList|.
+                      </li>
+                      <li>If |canonicalIndex| + 1 is greater than
+                      |buttonsSize|, then set |buttonsSize| to |canonicalIndex|
+                      + 1.
+                      </li>
+                    </ol>
+                  </li>
+                </ol>
+                <p>
+                  Otherwise, [=list/append=] |rawInputIndex| to
+                  |unmappedInputList|.
+                </p>
+              </li>
+              <li>Increment |rawInputIndex|.
+              </li>
+            </ol>
+          </li>
+          <li>Initialize |buttonIndex:long| to be 0.
+          </li>
+          <li>[=list/For each=] |rawInputIndex:long| of |unmappedInputList|:
+            <ol>
+              <li>While |mappedIndexList| [=list/contain=]s |buttonIndex|:
+                <ol>
+                  <li>Increment |buttonIndex|.
+                  </li>
+                </ol>
+              </li>
+              <li>Set |gamepad|.{{Gamepad/[[buttonMapping]]}}[|rawInputIndex|]
+              to |buttonIndex|.
+              </li>
+              <li>[=list/Append=] |buttonIndex| to |mappedIndexList|.
+              </li>
+              <li>If |buttonIndex| + 1 is greater than |buttonsSize|, then set
+              |buttonsSize| to |buttonIndex| + 1.
+              </li>
+            </ol>
+          </li>
+          <li>Initialize |buttons| to be an empty [=list=].
+          </li>
+          <li>[=list/For each=] |buttonIndex:long| of [=the range=] from 0 to
+          |buttonsSize| − 1, [=list/append=] a [=new=] {{GamepadButton}} to
+          |buttons|.
+          </li>
+          <li>Return |buttons|.
+          </li>
+        </ol>
+        <p>
+          User agents implementing this specification must provide a new DOM
+          event, named {{gamepadconnected}}. The corresponding event MUST be of
+          type {{GamepadEvent}} and, if [=allowed to use=] the `"gamepad"`
+          permission, MUST fire on the {{Window}} object. Registration for and
+          firing of the {{gamepadconnected}} event MUST follow the usual
+          behavior of DOM Events. [[DOM]]
+        </p>
+        <p>
+          A [=user agent=] MUST dispatch this event type to indicate the user
+          has connected a gamepad. If a gamepad was already connected when the
+          page was loaded, the {{gamepadconnected}} event SHOULD be dispatched
+          when the user presses a button or moves an axis.
+        </p>
+        <p>
+          A [=user agent=] MUST NOT dispatch this event type if the
+          [=environment settings object=] is a [=non-secure context=].
+        </p>
```

Comment:
We don't need this stuff anymore 🎉... it's already above (including the secure context stuff... the API is not exposed in insecure contexts, so that can't ever happen). 

```suggestion
```

Reply:
Oops, copy/paste error... removed